### PR TITLE
[WIP] Calendar report

### PR DESCRIPTION
**Pull request by lelemm**
A calendar view report:

https://github.com/user-attachments/assets/5a86b2fc-49fc-4acb-8d8d-3b7385251644

### CODE REVIEW

**Comment by lelemm** [comment]
This uses useTransactions added on PR #3685 from @joel-jeremy

**Comment by psybers** [comment]
Early feedback from manual testing:

- [ ] month names in the calendar widget should left align, to avoid crowing the blue/red amounts on the right
- [ ] clicking a day filters to that date, what about clicking the month name to filter to the month?
- [ ] on the dashboard, month names tend to wrap
- [ ] amounts are shown incorrectly, adding '00' to the amount. E.g. "52.23" becomes "5223.00"
- [ ] not able to reproduce at the moment, but somehow I went to 'all time' and the calendars were basically empty (except for December 2023).  Then I clicked to 3 months and it fixed itself.
- [ ] big black gap in the middle below the calendars:
image

**Comment by lelemm** [comment]
> * [x]  month names in the calendar widget should left align, to avoid crowing the blue/red amounts on the right
> * [x]  clicking a day filters to that date, what about clicking the month name to filter to the month?
> * [x]  on the dashboard, month names tend to wrap
> * [x]  amounts are shown incorrectly, adding '00' to the amount. E.g. "52.23" becomes "5223.00"

**Comment by psybers** [comment]
@lelemm Nice, thanks for the fixes!  I still see the month name not being left-aligned (it is centered, in the button).  Though I think you tweaked the spacing so now it is not as crowded. I would still recommend left aligning it as it feels a bit off as is.

I'm not sure I like the ':' in the month name on the dashboard.  Is it really needed?

BTW I get a crash now.  I set the view to 6 months, then clicked 'save' and tried to go back to the dashboard.

```
TypeError: Cannot read properties of undefined (reading 'length') at https://deploy-preview-3758.demo.actualbudget.org/static/js/ReportRouter.ydg2iLpI.chunk.js:3:75310 at Array.reduce (<anonymous>) at https://deploy-preview-3758.demo.actualbudget.org/static/js/ReportRouter.ydg2iLpI.chunk.js:3:75284 at Vy (https://deploy-preview-3758.demo.actualbudget.org/static/js/index.tOAhwMza.js:59:24270) at Uu (https://deploy-preview-3758.demo.actualbudget.org/static/js/index.tOAhwMza.js:59:42396) at https://deploy-preview-3758.demo.actualbudget.org/static/js/index.tOAhwMza.js:59:40713 at E (https://deploy-preview-3758.demo.actualbudget.org/static/js/index.tOAhwMza.js:44:1552) at MessagePort.L (https://deploy-preview-3758.demo.actualbudget.org/static/js/index.tOAhwMza.js:44:1914)
```

File: packages/desktop-client/src/components/accounts/Account.tsx [hunk 1]
```diff
@@ -146,7 +150,7 @@ type AllTransactionsProps = {
   collapseTransactions: (ids: string[]) => void;
 };
 
-function AllTransactions({
+export function AllTransactions({
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/accounts/Account.tsx:153]: Warning: this component is about the be removed in one of the in progress PRs: https://github.com/actualbudget/actual/pull/3708/files#diff-99c276500045bfda8f9c14b3c10705730a1fe7452a69246d4e40d092c24d305eL149

Comment (joel-jeremy) [reply packages/desktop-client/src/components/accounts/Account.tsx:153]: See https://github.com/actualbudget/actual/pull/3708/files#diff-99c276500045bfda8f9c14b3c10705730a1fe7452a69246d4e40d092c24d305eR508 instead.

**Comment by lelemm** [comment]
> @lelemm Nice, thanks for the fixes! I still see the month name not being left-aligned (it is centered, in the button). Though I think you tweaked the spacing so now it is not as crowded. I would still recommend left aligning it as it feels a bit off as is.
> 
> I'm not sure I like the ':' in the month name on the dashboard. Is it really needed?
> 
> BTW I get a crash now. I set the view to 6 months, then clicked 'save' and tried to go back to the dashboard.
> 
> ```
> TypeError: Cannot read properties of undefined (reading 'length') at https://deploy-preview-3758.demo.actualbudget.org/static/js/ReportRouter.ydg2iLpI.chunk.js:3:75310 at Array.reduce (<anonymous>) at https://deploy-preview-3758.demo.actualbudget.org/static/js/ReportRouter.ydg2iLpI.chunk.js:3:75284 at Vy (https://deploy-preview-3758.demo.actualbudget.org/static/js/index.tOAhwMza.js:59:24270) at Uu (https://deploy-preview-3758.demo.actualbudget.org/static/js/index.tOAhwMza.js:59:42396) at https://deploy-preview-3758.demo.actualbudget.org/static/js/index.tOAhwMza.js:59:40713 at E (https://deploy-preview-3758.demo.actualbudget.org/static/js/index.tOAhwMza.js:44:1552) at MessagePort.L (https://deploy-preview-3758.demo.actualbudget.org/static/js/index.tOAhwMza.js:44:1914)
> ```

nice catch. 
- removed the :
- fixed the crash
- changed from centered to left aligned

File: packages/desktop-client/src/components/accounts/Account.tsx (hunk 1 above)
Comment (lelemm) [reply packages/desktop-client/src/components/accounts/Account.tsx:153]: oh, I added this today cuz to fix the missing split transactions.
can't find the changes you referenced.
whats the name of the file so I can filter it?

Comment (lelemm) [reply packages/desktop-client/src/components/accounts/Account.tsx:153]: got my way in

Comment (lelemm) [reply packages/desktop-client/src/components/accounts/Account.tsx]: fixed

Comment (lelemm) [reply packages/desktop-client/src/components/accounts/Account.tsx]: btw, I will have to go back to this transaction list to make it more functional

**Comment by jsehnoutka** [comment]
This is probably not relevant for the initial build, but I see the most potential using this for month-based planning as outlined in #1928

- Future schedule-based transactions could be indicated , ideally in the enhanced range introduced by #3310
- Could also show running balance for each day when filtered by account (also for future expenses) to enable for transfer planning between accounts

In any case, thanks for this, great feature!

**Comment by lelemm** [comment]
> Edit: Found an inconsistency between symbols for incomes and expenses once the amount goes through xx xxx,xx amount
> 
> image image image

I've noticed that too. when there is no space available it is shrinking the icon

**Comment by joel-jeremy** [comment]
> > Edit: Found an inconsistency between symbols for incomes and expenses once the amount goes through xx xxx,xx amount
> > image image image
> 
> I've noticed that too. when there is no space available it is shrinking the icon

You can set flexShrink: 0 on the icon style to prevent it from shrinking

**Comment by lelemm** [comment]
> * [ ]  month names in the calendar widget should left align, to avoid crowing the blue/red amounts on the right
> * [ ]  clicking the 'select all' checkbox at the top of the transactions does nothing

- Changed the way month names work and also aligned to left
- Removed checkboxes from the transaction list in the report details (removed split transaction operation too)

**Comment by lelemm** [comment]
> This is probably not relevant for the initial build, but I see the most potential using this for month-based planning as outlined in #1928
> 
> * Future schedule-based transactions could be indicated , ideally in the enhanced range introduced by [Add option to set how far out the upcoming scheduled transactions are shown in the account view. #3310](https://github.com/actualbudget/actual/pull/3310)
> * Could also show running balance for each day when filtered by account (also for future expenses) to enable for transfer planning between accounts
> 
> In any case, thanks for this, great feature!
> 
> Edit: Found an inconsistency between symbols for incomes and expenses once the amount goes through xx xxx,xx amount
> 
> image image image

solved

**Comment by lelemm** [comment]
Added mobile page details:

image

**Comment by lelemm** [comment]
changed this back to WIP since it depends on useTransactions from @joel-jeremy and its not merged yet

**Pull request closed without merging**